checks/remediations.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checks
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/clients"
+)
+
+var (
+	remediationBranch   string
+	remediationRepo     string
+	remediationOnce     *sync.Once
+	remediationSetupErr error
+)
+
+//nolint:gochecknoinits
+func init() {
+	remediationOnce = new(sync.Once)
+}
+
+func remdiationSetup(c *checker.CheckRequest) error {
+	remediationOnce.Do(func() {
+		// Get the branch for remediation.
+		b, err := c.RepoClient.GetDefaultBranch()
```

Comment:
Should this be the default branch or the branch on which Scorecard Action was run? As an example, if one creates a PR with a new workflow, which is missing permissions, this will still have main (/default branch). But when app.stepsecurity.io tries to fix it, it will not find the workflow in that branch.

Reply:
You're correct. However we only support the default branch for non pull_request events https://github.com/ossf/scorecard-action/blob/main/entrypoint.sh#L101 and we don't officially support (https://github.com/ossf/scorecard-action below the table of content).

So I think we only need to support the default branch for now. Let me know if I missed some cases, it's possible

Reply:
ok, thanks for the info!